src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -1,36 +1,299 @@
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractFile;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractRequestDetailFileLine;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestedCashAdvance;
 import edu.cornell.kfs.concur.batch.service.ConcurCashAdvancePdpFeedFileService;
+import edu.cornell.kfs.concur.batch.service.ConcurRequestedCashAdvanceService;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedAccountingEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedDetailEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedFileBaseEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedGroupEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedHeaderEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedPayeeIdEntry;
+import edu.cornell.kfs.concur.batch.xmlObjects.PdpFeedTrailerEntry;
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+import edu.cornell.kfs.sys.service.CUMarshalService;
 
-public class ConcurCashAdvancePdpFeedFileServiceImpl extends InitiateDirectoryBase implements ConcurCashAdvancePdpFeedFileService {
+public class ConcurCashAdvancePdpFeedFileServiceImpl implements ConcurCashAdvancePdpFeedFileService {
     private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurCashAdvancePdpFeedFileServiceImpl.class);
-    protected String outgoingDirectoryName;
+    private int payeeNameFieldSize = 0;
+    protected String paymentImportDirectory;
+    protected DateTimeService DateTimeService;
+    protected DataDictionaryService dataDictionaryService;
+    protected ConcurRequestedCashAdvanceService concurRequestedCashAdvanceService;
+    protected CUMarshalService cuMarshalService;
+    protected ParameterService parameterService;
+
+    @Transactional
+    public boolean createPdpFeedFileForValidatedDetailFileLines(ConcurRequestExtractFile requestExtractFile) {
+        boolean pdpFileSuccessfullyCreated = false;
+        PdpFeedFileBaseEntry pdpFeedFileDataObject = buildPdpFeedBaseEntry(requestExtractFile);
+        if (pdpFeedFileDataObject.getTrailer().getDetailCount().intValue() != 0) {
+            String fullyQualifiedPdpFileName = buildFullyQualifiedPdpOutputFileName(getPaymentImportDirectory(), requestExtractFile.getFileName());
+            pdpFileSuccessfullyCreated = createPdpFeedFile(pdpFeedFileDataObject, fullyQualifiedPdpFileName);
+            if (pdpFileSuccessfullyCreated) {
+                LOG.debug("fullyQualifiedPdpFileName [" + fullyQualifiedPdpFileName + "]  was created for requestExtractFile [" + requestExtractFile.getFileName() + "]");
+                requestExtractFile.setFullyQualifiedPdpFileName(fullyQualifiedPdpFileName);
+            }
+            else {
+                LOG.debug("FAILED TO CREATE: fullyQualifiedPdpFileName [" + fullyQualifiedPdpFileName + "] for requestExtractFile [" + requestExtractFile.getFileName() + "]");
```

Comment:
Changing from LOG.debug to LOG.error and changing the "if" LOG.debug to LOG.info